python/paddle/distributed/auto_parallel/static/parallelizer_v2.py

```diff
@@ -428,8 +429,21 @@ def _apply_post_optimization(
             and self._strategy.pipeline.enable
             and use_new_executor()
         ):
+            enable_send_recv_overlap = (
+                self._strategy.pipeline.enable_send_recv_overlap
+            )
+            if (
+                enable_send_recv_overlap
+                and int(os.getenv("CUDA_DEVICE_MAX_CONNECTIONS", "0")) != 1
+            ):
+                self._logger.warning(
+                    "enable_send_recv_overlap requires environment variable CUDA_DEVICE_MAX_CONNECTIONS=1, but you did not set it. Paddle has been automatically set CUDA_DEVICE_MAX_CONNECTIONS to 1, try export CUDA_DEVICE_MAX_CONNECTIONS=1 to make this warning disappear."
```

Comment:
This env var may reduce the overall performance?

Reply:
No large-scale testing yet. It has been changed to a WARNING, consistent with the behavior of the dynamic graph:
https://github.com/PaddlePaddle/Paddle/blob/a7ad947eb4a24fb838b2f57ccc6c7ea4f72c2222/python/paddle/distributed/fleet/layers/mpu/mp_layers.py#L230